wazuh/wazuh-splunk
# Improve packages' support for minor and patch versions of Splunk

## Summary

Closes #1308 

For details about the development head to the related issue.

![splunk_versions](https://user-images.githubusercontent.com/15186973/168097692-05b635d2-d18e-42ba-8162-fdb671ff804b.gif)

### To test
- Check that changing the Wazuh's version, Splunk's version and App's revision on the `package.json` and running the pre-release script using the `make prebuild` commnand changes the related information on the `app.conf`, `package.conf` and `appVersionService.js` files, as seen on the gif above.
- Check that the logic to check if the App revisions on the frontend and the backend match works as expected. For this, do the following:
  - Change the App's revision on the `appVersionService.js` file.
  - Go to the main page of Splunk (exit the App)
  - Refresh the page and enter the Wazuh app again.
  - A warning toast should show up indicating that the App revision numbers do not match.
  ![image](https://user-images.githubusercontent.com/15186973/168115277-07119c4e-17eb-44f7-af77-1b896c79a999.png)
- Check that the logic to check if the App version and the Wazuh API version match work as expected. For this, do the following:
  - Revert any previous change.
  - Change the Wazuh version on the `app.conf` file.
  - Open the App.
  - A warning toast should show up indicating that the versions of Wazuh from the App and the API do not match.
![image](https://user-images.githubusercontent.com/15186973/168134576-8c1854bb-0a77-448b-bf46-5eccd58b0b30.png)
- Check that no API can be added if its version do not match the App's version. 
  - To test this, change the Wazuh version on the `app.conf` and try to add a new API from the frontend (its version must be different that the one specified on the `app.conf`).
  - Check that the API is not added and an error toast shows up.
  - Go to the Logs section and check that the error is due to different Wazuh versions of the API and the App.
![image](https://user-images.githubusercontent.com/15186973/168323841-4e8de3e1-918e-4494-b3fd-3343815c8d85.png)
